### PR TITLE
Refactor arrow kernels to include valueBuilder

### PR DESCRIPTION
**Pull request by fedor-miron**
### Changelog entry 

Refactor arrow kernels to include another argument (valueBuilder). It is required by Datetime UDF.

### Additional information

...

### CODE REVIEW

File: ydb/library/yql/public/udf/arrow/udf_arrow_helpers.h [hunk 1]
```diff
@@ -377,8 +383,8 @@ struct TUnaryKernelExec {
             for (int64_t i = 0; i < array.length;) {
                 for (size_t j = 0; j < maxBlockLength && i < array.length; ++j, ++i) {
                     auto item = readerImpl->GetItem(array, i);
-                    TDerived::Process(item, [&](TBlockItem out) {
-                        builderImpl->Add(out);
+                    TDerived::Process(item, state.GetValueBuilder(), [&](auto out) {
+                        builder.Add(out);
```
Comment (igormunkin) [inline ydb/library/yql/public/udf/arrow/udf_arrow_helpers.h:387]: Fixup was reverted in scope of bdd9c882ca5512cb5cdf36c39cbb942d20a640eb.

**Review by igormunkin — CHANGES_REQUESTED** [review]
@fedor-miron, thanks for the patch! Please, consider my comments below.

**Review by igormunkin — DISMISSED** [review]
@fedor-miron, thanks for the fixes! LGTM, now.

File: ydb/library/yql/udfs/common/url_base/lib/url_base_udf.h [hunk 2]
```diff
@@ -36,7 +36,7 @@ inline bool PrepareUrl(const std::string_view& keyStr, TUri& parser) {
     } \
     struct udfName##KernelExec : public TUnaryKernelExec<udfName##KernelExec> { \
         template <typename TSink> \
-        static void Process(TBlockItem arg, const TSink& sink) { \
+        static void Process(TBlockItem arg, const IValueBuilder&, const TSink& sink) { \
```
Comment (vitstn) [inline ydb/library/yql/udfs/common/url_base/lib/url_base_udf.h:39]: ValueBuilder должен быть первым аргументом, как в Run

Comment (vitstn) [reply ydb/library/yql/udfs/common/url_base/lib/url_base_udf.h:39]: и указателем

**Review by vitstn — CHANGES_REQUESTED** [review]
issues

**Review by vitstn — APPROVED** [review]

**Pull request merged**